src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -574,10 +574,7 @@ internal static void EmitConstant(this ILGenerator il, object value, Type type)
             throw ContractUtils.Unreachable;
         }
 
-        internal static bool ShouldLdtoken(Type t)
-        {
-            return t is TypeBuilder || t.IsGenericParameter || t.GetTypeInfo().IsVisible;
-        }
+        private static bool ShouldLdtoken(Type t) => t.IsGenericParameter || t.GetTypeInfo().IsVisible;
```

Comment:
Are there any situations in which the t is TypeBuilder clause is valid?  I'm wondering if the right answer is to remove it or if the right answer is to further qualify it?